src/main/java/org/opensearch/neuralsearch/processor/NeuralSparseTwoPhaseProcessor.java

```diff
@@ -0,0 +1,223 @@
+
+            return new NeuralSparseTwoPhaseProcessor(tag, description, ignoreFailure, enabled, ratio, window_expansion, max_window_size);
+        }
+    }
+
+    private QueryBuilder getNestedQueryBuilderFromNeuralSparseQueryBuilderMap(
+        final Multimap<NeuralSparseQueryBuilder, Float> queryBuilderFloatMap
+    ) {
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        queryBuilderFloatMap.asMap().forEach((neuralSparseQueryBuilder, boosts) -> {
+            float reduceBoost = boosts.stream().reduce(0.0f, Float::sum);
+            final float finalBoost = neuralSparseQueryBuilder.boost() * reduceBoost;
+            boolQueryBuilder.should(neuralSparseQueryBuilder.boost(finalBoost));
+        });
+        return boolQueryBuilder;
+    }
+
+    private float getOriginQueryWeightAfterRescore(final SearchSourceBuilder searchSourceBuilder) {
+        if (searchSourceBuilder.rescores() == null) return 1.0f;
+        return searchSourceBuilder.rescores()
+            .stream()
+            .map(rescorerBuilder -> ((QueryRescorerBuilder) rescorerBuilder).getQueryWeight())
+            .reduce(1.0f, (a, b) -> a * b);
+    }
+
+    private Multimap<NeuralSparseQueryBuilder, Float> collectNeuralSparseQueryBuilder(final QueryBuilder queryBuilder, float baseBoost) {
+        Multimap<NeuralSparseQueryBuilder, Float> result = ArrayListMultimap.create();
+
+        if (queryBuilder instanceof BoolQueryBuilder) {
+            BoolQueryBuilder boolQueryBuilder = (BoolQueryBuilder) queryBuilder;
+            float updatedBoost = baseBoost * boolQueryBuilder.boost();
+            for (QueryBuilder subQuery : boolQueryBuilder.should()) {
+                Multimap<NeuralSparseQueryBuilder, Float> subResult = collectNeuralSparseQueryBuilder(subQuery, updatedBoost);
+                result.putAll(subResult);
+            }
+        } else if (queryBuilder instanceof NeuralSparseQueryBuilder) {
+            NeuralSparseQueryBuilder neuralSparseQueryBuilder = (NeuralSparseQueryBuilder) queryBuilder;
+            float updatedBoost = baseBoost * neuralSparseQueryBuilder.boost();
+            NeuralSparseQueryBuilder modifiedQueryBuilder = neuralSparseQueryBuilder.getCopyNeuralSparseQueryBuilderForTwoPhase(ratio);
+            result.put(modifiedQueryBuilder, updatedBoost);
+        }
+        // We only support BoostQuery, BooleanQuery and NeuralSparseQuery now. For other compound query type which are not support now, will
+        // do nothing and just quit.
+        return result;
+    }
+
+    private RescorerBuilder<QueryRescorerBuilder> buildRescoreQueryBuilderForTwoPhase(final QueryBuilder nestedTwoPhaseQueryBuilder, final SearchRequest searchRequest){
+        RescorerBuilder<QueryRescorerBuilder> twoPhaseRescorer = new QueryRescorerBuilder(nestedTwoPhaseQueryBuilder);
+        int requestSize = searchRequest.source().size();
+        int windowSize = (int) ((requestSize == -1 ? 10 : requestSize) * window_expansion);
```

Comment:
Looks like 10 is a magic number, making it a constant and add comments to it.

Reply:
Not a magic number, it's the default value of a query in opensearch now, make it to constant now.